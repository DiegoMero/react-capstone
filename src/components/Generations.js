import './style/Generations.css';
import { Link } from 'react-router-dom';
import kanto from './assets/kanto.png';
import johto from './assets/johto.png';
import hoenn from './assets/hoenn.png';
import sinnoh from './assets/sinnoh.png';
import teselia from './assets/teselia.png';
import kalos from './assets/kalos.png';
import alola from './assets/alola.png';
import galar from './assets/galar.png';

const Generations = () => (
  <div className="generations-section">
    <h2>Pokemon Generations</h2>
    <div className="generations">
      <Link to="/Generation1">
        <img src={kanto} alt="Kanto" className="region" />
        <h3>1st Generation</h3>
        <h4>Kanto</h4>
      </Link>
      <Link to="/Generation2">
        <img src={johto} alt="Johto" className="region" />
        <h3>2st Generation</h3>
        <h4>Johto</h4>
      </Link>
      <Link to="/Generation3">
        <img src={hoenn} alt="Kanto" className="region" />
        <h3>3rd Generation</h3>
        <h4>Hoenn</h4>
      </Link>
      <Link to="/Generation4">
        <img src={sinnoh} alt="Kanto" className="region" />
        <h3>4th Generation</h3>
        <h4>Sinnoh</h4>
      </Link>
      <Link to="/Generation5">
        <img src={teselia} alt="Kanto" className="region" />
        <h3>5th Generation</h3>
        <h4>Teselia</h4>
      </Link>
      <Link to="/Generations6">
        <img src={kalos} alt="Kanto" className="region" />
        <h3>6th Generation</h3>
        <h4>Kalos</h4>
      </Link>
      <Link to="/Generations7">
        <img src={alola} alt="Kanto" className="region" />
        <h3>7th Generation</h3>
        <h4>Alola</h4>
      </Link>
      <Link to="/Generations8">
        <img src={galar} alt="Kanto" className="region" />
        <h3>8th Generation</h3>
        <h4>Galar</h4>
      </Link>
    </div>
  </div>
);

export default Generations;

/* const pokemonList = useSelector((state) => state.pokemon);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!pokemonList.length) dispatch(loadPokemon1G());
  }, []);

  return (
    <ul>
      {pokemonList.map((pokemon) => (
        <li key={uuidv4()}>{pokemon.pokemonName}</li>
      ))}
    </ul>
  ); */
