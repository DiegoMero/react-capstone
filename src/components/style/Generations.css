.generations {
  display: grid;
  grid-template-columns: 50% 50%;
}

.region {
  width: 160px;
  height: 160px;
}

.region-stats {
  background-color: rgb(221, 29, 103);
  color: white;
  margin: 0;
}

.title {
  background-color: rgb(216, 81, 132);
  margin: 0;
  color: white;
  padding: 10px;
  text-align: center;
}

.pokemon-species {
  text-align: center;
  background-color: rgb(216, 81, 132);
  margin: 0;
  padding: 10px;
  color: white;
}

a {
  text-decoration: none;
}

.kanto-link {
  background-color: rgb(221, 29, 103);
}

.johto-link {
  background-color: rgb(216, 81, 132);
}

.hoenn-link {
  background-color: rgb(216, 81, 132);
}

.sinnoh-link {
  background-color: rgb(221, 29, 103);
}

.teselia-link {
  background-color: rgb(221, 29, 103);
}

.kalos-link {
  background-color: rgb(216, 81, 132);
}

.alola-link {
  background-color: rgb(216, 81, 132);
}

.galar-link {
  background-color: rgb(221, 29, 103);
}

.region-name {
  color: white;
  margin: 0;
  text-align: end;
  padding: 5px;
}