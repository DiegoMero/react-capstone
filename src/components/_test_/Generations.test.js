import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import Generations from '../Generations';
import store from '../../redux/configureStore';

test('snapshot for Header', () => {
  const tree = render(
    <React.StrictMode>
      <BrowserRouter>
        <Provider store={store}>
          <Generations />
        </Provider>
      </BrowserRouter>
    </React.StrictMode>,
  );
  expect(tree).toMatchSnapshot();
});
