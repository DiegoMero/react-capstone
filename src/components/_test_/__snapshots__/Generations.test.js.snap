// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="generations-section"
      >
        <h1
          class="title"
        >
          Pokemon World
        </h1>
        <h2
          class="pokemon-species"
        >
          905 Pokemon Species
        </h2>
        <h4
          class="region-stats"
        >
          SPECIES BY REGION
        </h4>
        <div
          class="generations"
        >
          <a
            class="kanto-link"
            href="/Generation1"
          >
            <img
              alt="Kanto"
              class="region"
              src="kanto.png"
            />
            <h3
              class="region-name"
            >
              KANTO
            </h3>
            <h3
              class="region-name"
            >
              -754 species
            </h3>
          </a>
          <a
            class="johto-link"
            href="/Generation2"
          >
            <img
              alt="Johto"
              class="region"
              src="johto.png"
            />
            <h3
              class="region-name"
            >
              JOHTO
            </h3>
            <h3
              class="region-name"
            >
              -805 species
            </h3>
          </a>
          <a
            class="hoenn-link"
            href="/Generation3"
          >
            <img
              alt="Hoenn"
              class="region"
              src="hoenn.png"
            />
            <h3
              class="region-name"
            >
              HOENN
            </h3>
            <h3
              class="region-name"
            >
              -770 species
            </h3>
          </a>
          <a
            class="sinnoh-link"
            href="/Generation4"
          >
            <img
              alt="Sinnoh"
              class="region"
              src="sinnoh.png"
            />
            <h3
              class="region-name"
            >
              SINNOH
            </h3>
            <h3
              class="region-name"
            >
              -798 species
            </h3>
          </a>
          <a
            class="teselia-link"
            href="/Generation5"
          >
            <img
              alt="Teselia"
              class="region"
              src="teselia.png"
            />
            <h3
              class="region-name"
            >
              TESELIA
            </h3>
            <h3
              class="region-name"
            >
              -749 species
            </h3>
          </a>
          <a
            class="kalos-link"
            href="/Generation6"
          >
            <img
              alt="Kalos"
              class="region"
              src="kalos.png"
            />
            <h3
              class="region-name"
            >
              KALOS
            </h3>
            <h3
              class="region-name"
            >
              -833 species
            </h3>
          </a>
          <a
            class="alola-link"
            href="/Generation7"
          >
            <img
              alt="Alola"
              class="region"
              src="alola.png"
            />
            <h3
              class="region-name"
            >
              ALOLA
            </h3>
            <h3
              class="region-name"
            >
              -817 species
            </h3>
          </a>
          <a
            class="galar-link"
            href="/Generation8"
          >
            <img
              alt="Galar"
              class="region"
              src="galar.png"
            />
            <h3
              class="region-name"
            >
              GALAR
            </h3>
            <h3
              class="region-name"
            >
              -809 species
            </h3>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="generations-section"
    >
      <h1
        class="title"
      >
        Pokemon World
      </h1>
      <h2
        class="pokemon-species"
      >
        905 Pokemon Species
      </h2>
      <h4
        class="region-stats"
      >
        SPECIES BY REGION
      </h4>
      <div
        class="generations"
      >
        <a
          class="kanto-link"
          href="/Generation1"
        >
          <img
            alt="Kanto"
            class="region"
            src="kanto.png"
          />
          <h3
            class="region-name"
          >
            KANTO
          </h3>
          <h3
            class="region-name"
          >
            -754 species
          </h3>
        </a>
        <a
          class="johto-link"
          href="/Generation2"
        >
          <img
            alt="Johto"
            class="region"
            src="johto.png"
          />
          <h3
            class="region-name"
          >
            JOHTO
          </h3>
          <h3
            class="region-name"
          >
            -805 species
          </h3>
        </a>
        <a
          class="hoenn-link"
          href="/Generation3"
        >
          <img
            alt="Hoenn"
            class="region"
            src="hoenn.png"
          />
          <h3
            class="region-name"
          >
            HOENN
          </h3>
          <h3
            class="region-name"
          >
            -770 species
          </h3>
        </a>
        <a
          class="sinnoh-link"
          href="/Generation4"
        >
          <img
            alt="Sinnoh"
            class="region"
            src="sinnoh.png"
          />
          <h3
            class="region-name"
          >
            SINNOH
          </h3>
          <h3
            class="region-name"
          >
            -798 species
          </h3>
        </a>
        <a
          class="teselia-link"
          href="/Generation5"
        >
          <img
            alt="Teselia"
            class="region"
            src="teselia.png"
          />
          <h3
            class="region-name"
          >
            TESELIA
          </h3>
          <h3
            class="region-name"
          >
            -749 species
          </h3>
        </a>
        <a
          class="kalos-link"
          href="/Generation6"
        >
          <img
            alt="Kalos"
            class="region"
            src="kalos.png"
          />
          <h3
            class="region-name"
          >
            KALOS
          </h3>
          <h3
            class="region-name"
          >
            -833 species
          </h3>
        </a>
        <a
          class="alola-link"
          href="/Generation7"
        >
          <img
            alt="Alola"
            class="region"
            src="alola.png"
          />
          <h3
            class="region-name"
          >
            ALOLA
          </h3>
          <h3
            class="region-name"
          >
            -817 species
          </h3>
        </a>
        <a
          class="galar-link"
          href="/Generation8"
        >
          <img
            alt="Galar"
            class="region"
            src="galar.png"
          />
          <h3
            class="region-name"
          >
            GALAR
          </h3>
          <h3
            class="region-name"
          >
            -809 species
          </h3>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
